cypress/e2e/GUI/gui.cy.js

```diff
@@ -0,0 +1,141 @@
+/// <reference types="cypress" />
+describe('GUI Tests', () => {
+    beforeEach(function () {
+        cy.visit('/')
+        cy.contains('button', 'Accept').click()
+    })
+
+    it('Mostra as colunas Nome da Empresa e Acao, e oculta as colunas ID, Industria, Numero de Funcionarios e Tamanho em um viewport movel', () => {
+        cy.viewport(320, 480)
+        cy.contains('Company name').should('be.visible')
+        cy.contains('Action').should('be.visible')
+        cy.contains('Industry').should('not.be.visible')
+        cy.contains('Number of employees').should('not.be.visible')
+        cy.contains('Size').should('not.be.visible')
+    })
+
+    it('Exibe o cabecalho com um titulo, alternador de tema e um campo de entrada de texto', () => {
+        cy.contains('EngageSphere').should('be.visible')
+        cy.get('[class^="ThemeToggle_container"]').should('be.visible')
+        cy.get('input[id="name"]').should('be.visible')
+    })
+
+    it('Exibe a saudacao "Hi there" quando nenhum nome e fornecido', () => { 
+        cy.contains('h2', 'Hi there!').should('be.visible')
+    })
+
+    it('Exibe a saudacao "Hi Joe" quando o nome e fornecido', () => {
+        cy.get('input[data-testid="name"]').type('Joe')
+        cy.contains('h2', 'Hi Joe!').should('be.visible')
+    })
+
+    it('Exibe o rodape com o texto e links corretos', () => {
+        cy.contains('p', 'Copyright 2024 - Talking About Testing')
+            .should('be.visible')
+        cy.contains('a', 'Hotmart')
+            .should('be.visible')
+            .and('have.attr', 'href', 'https://hotmart.com/pt-br/club/cypress-playground-ate-a-nuvem')
+        cy.contains('a', 'Udemy')
+            .should('be.visible')
+            .and('have.attr', 'href', 'https://udemy.com/user/walmyr')
+        cy.contains('a', 'Blog')
+            .should('be.visible')
+            .and('have.attr', 'href', 'https://talkingabouttesting.com')
+        cy.contains('a', 'YouTube')
+            .should('be.visible')
+            .and('have.attr', 'href', 'https://youtube.com/@talkingabouttesting')
+    })
+
+    it('Verifica se e possivel alterar entre os temas corretamente', () => {
+        cy.get('[class^="ThemeToggle_container"]').click()
+        cy.get('body[data-theme="dark"]').should('be.visible')
+        cy.get('[class^="ThemeToggle_container"]').click()
+        cy.get('body[data-theme="light"]').should('be.visible')
+    })
+
+    it('Mantem os filtros ao voltar da visualizacao de detalhes do cliente', () => {
+        cy.get('select[data-testid="size-filter"]').select('Small')
+        cy.get('select[data-testid="industry-filter"]').select('Logistics')
+        cy.contains('button', 'View').click()
+        cy.contains('Customer Details').should('be.visible')
+        cy.contains('button', 'Back').click()
```

Comment:
```suggestion
        cy.contains('button', 'Back').click()

```

Só separando as pré-condições, ações e resultados esperados.

Falarei sobre isso no último encontro da _Test Design Masterclass_ ao abordar o padrão AAA (_Arrange, Act, Assert_).